api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -177,6 +191,8 @@ private static Object internalFromByteBuffer(Type type, ByteBuffer buffer) {
         byte[] unscaledBytes = new byte[buffer.remaining()];
         tmp.get(unscaledBytes);
         return new BigDecimal(new BigInteger(unscaledBytes), decimal.scale());
+      case VARIANT:
+        return Variant.from(buffer);
```

Comment:
nit: shall we use `tmp` instead of `buffer` to honor the LE order that was set above?